src/common/Logging/MacAddressHash/MACInformationProvider.cs

```diff
@@ -157,34 +156,19 @@ private void PersistMacAddressHash(string hashedMacAddress)
         private static bool ValidateMacAddressHash(string macAddressHash)
             => !string.IsNullOrEmpty(macAddressHash) && Regex.IsMatch(macAddressHash, PersistRegex);
 
-        private static string RunCommandAndGetOutput(string commandName, string commandArgs = null)
+        private string RunCommandAndGetOutput(string commandName, string commandArgs = null)
         {
-            var processOutput = new StringBuilder();
-            var process = new Process();
             try
             {
-                process.EnableRaisingEvents = true;
-                process.StartInfo.UseShellExecute = false;
-                process.StartInfo.CreateNoWindow = true;
-                process.StartInfo.RedirectStandardOutput = true;
-
-                process.StartInfo.RedirectStandardError = true;
-                process.StartInfo.FileName = commandName;
-                process.StartInfo.Arguments = commandArgs ?? string.Empty;
-                process.OutputDataReceived += (object sender, DataReceivedEventArgs e) =>
-                {
-                    processOutput.AppendLine(e.Data);
-                };
-                process.Start();
-                process.BeginOutputReadLine();
-                process.BeginErrorReadLine();
-                process.WaitForExit();
-                process.Close();
-                return processOutput.ToString();
+                (var exitCode, var output) = this._platform.ExecuteAndReturnOutput(commandName, 
+                                                                              commandArgs, 
+                                                                              timeout: TimeSpan.FromSeconds(30), 

```

Comment:
@Eneuman can you document or leave a comment why it was set to 30 seconds timeout please ? Thank you.

Reply:
@hsubramanianaks I changed the description in the PR :) Do you want me to add a comment to the code aswell?

Reply:
Thank you, description should be fine.